pyrit/ui/scorer.py

```diff
@@ -0,0 +1,101 @@
+from connection_status import ConnectionStatusHandler
+
+class GradioApp:
+    def __init__(self):
+        self.i = 0
+        self.rpc_client = RPCClient(self._disconnected_rpc_callback)
+        self.connect_status = None
+        self.url = ""
+    
+    def start_gradio(self, open_browser=False):
+        with gr.Blocks() as demo:
+            is_connected = gr.State(False)
+            converted_value_state = gr.State("")
+            
+            self.connect_status = ConnectionStatusHandler(is_connected, self.rpc_client)
+            with gr.Column(visible=False) as main_interface:
+                prompt = gr.Markdown("Prompt: ")
+                prompt.height = "200px"
+                with gr.Row():
+                    safe = gr.Button("Safe")
+                    unsafe = gr.Button("Unsafe")
+                    
+                    safe.click(
+                        fn=lambda: [gr.update(interactive=False)]*2 + [""],outputs=[safe, unsafe, converted_value_state]
+                    ).then(
+                        fn=self._safe_clicked, outputs=converted_value_state
+                    )
+                    unsafe.click(
+                        fn=lambda: [gr.update(interactive=False)]*2 + [""], outputs=[safe, unsafe, converted_value_state]
+                    ).then(
+                        fn=self._unsafe_clicked, outputs=converted_value_state
+                    )
+            
+            with gr.Row() as loading_animation:
+                loading_text = gr.Markdown("Connecting to PyRIT")
+                timer = gr.Timer(0.5)
+                timer.tick(fn=self._loading_dots, outputs=loading_text)
+
+            converted_value_state.change(fn=self._on_converted_value_change, inputs=[converted_value_state], outputs=[prompt, safe, unsafe])
+            self.connect_status.setup(
+                main_interface=main_interface,
+                loading_animation=loading_animation,
+                converted_value_state=converted_value_state)
+
+            demo.load(
+                fn=self._main_interface_loaded,
+                outputs=[main_interface, loading_animation, converted_value_state, is_connected]
+            )
+
+        if open_browser:
+            demo.launch(inbrowser=True)
+        else:
+            _, url, _ = demo.launch(prevent_thread_lock=True)
+            self.url = url
+            print("Gradio launched")
+            webview.create_window("PyRIT - Scorer", self.url)
+            webview.start()
+            print("Webview closed!")
+
+        if self.rpc_client:
+            self.rpc_client.stop()
+        
+    def _safe_clicked(self):
+        return self._send_prompt_response(True)
+
+    def _unsafe_clicked(self):
+        return self._send_prompt_response(False)
+
+    def _send_prompt_response(self, value):
+        self.rpc_client.send_prompt_response(value)
+        converted_value = self.rpc_client.wait_for_prompt()
+        return str(converted_value.original_value)
```

Comment:
sorry if I was unclear in my previous comments! `original_value` should be `converted_value` but the previous name of prompt can stay (i.e.
```suggestion
        prompt = self.rpc_client.wait_for_prompt()
        return str(prompt.converted_value)
```